src/databricks/labs/ucx/runtime.py

```diff
@@ -433,27 +433,42 @@ def migrate_external_tables_sync(
     - For Azure: `principal-prefix-access`, `create-table-mapping`, `create-uber-principal`, `migrate-credentials`, `migrate-locations`, `create-catalogs-schemas`
     - For AWS: TBD
     """
+    migrate_tables(cfg, ws, sql_backend, install, What.EXTERNAL_SYNC)
+
+
+@task("migrate-tables", job_cluster="table_migration")
+def migrate_dbfs_root_delta_tables(
+    cfg: WorkspaceConfig, ws: WorkspaceClient, sql_backend: SqlBackend, install: Installation
+):
+    """This workflow task migrates `delta tables stored in DBFS root` from the Hive Metastore to the Unity Catalog using deep clone.
+    Following cli commands are required to be run before running this task:
+    - For Azure: `principal-prefix-access`, `create-table-mapping`, `create-uber-principal`, `migrate-credentials`, `migrate-locations`, `create-catalogs-schemas`
+    - For AWS: TBD
+    """
+    migrate_tables(cfg, ws, sql_backend, install, What.DBFS_ROOT_DELTA)
+
+
+@task("migrate-tables", job_cluster="table_migration")
+def migrate_views(cfg: WorkspaceConfig, ws: WorkspaceClient, sql_backend: SqlBackend, install: Installation):
+    """This workflow task migrates `delta tables stored in DBFS root` from the Hive Metastore to the Unity Catalog using deep clone.
+    Following cli commands are required to be run before running this task:
+    - For Azure: `principal-prefix-access`, `create-table-mapping`, `create-uber-principal`, `migrate-credentials`, `migrate-locations`, `create-catalogs-schemas`
+    - For AWS: TBD
+    """
+    migrate_tables(cfg, ws, sql_backend, install, What.VIEW)
+
+
+def migrate_tables(
+    cfg: WorkspaceConfig, ws: WorkspaceClient, sql_backend: SqlBackend, install: Installation, what: What
+):
     table_crawler = TablesCrawler(sql_backend, cfg.inventory_database)
     udf_crawler = UdfsCrawler(sql_backend, cfg.inventory_database)
     grant_crawler = GrantsCrawler(table_crawler, udf_crawler)
     table_mappings = TableMapping(install, ws, sql_backend)
     migration_status_refresher = MigrationStatusRefresher(ws, sql_backend, cfg.inventory_database, table_crawler)
     group_manager = GroupManager(sql_backend, ws, cfg.inventory_database)
     mount_crawler = Mounts(sql_backend, ws, cfg.inventory_database)
-    cluster_locations = {}
-    if ws.config.is_azure:
-        locations = ExternalLocations(ws, sql_backend, cfg.inventory_database)
-        azure_client = AzureAPIClient(
```

Comment:
See message in Slack from @nfx about these calls only existing from the cli not within a job